Bugfix

mikhaillopandia opened the pull request.

Commit:
Bugfixes

Now it works fine

Commit:
Text waiting

Added rspec-wait feature for JS provided text waiting

Commit:
Works fine

Commit:
Rubocop fixes

Commit:
Scenarios have been changed

Added cucumber tables to scenarios
Transformations have been deleted

Commit:
Cosmetic corrections

Commit:
Rubocop corrections

romikoops commented on features/signup.feature:
```diff
@@ -6,76 +6,58 @@ Feature: Sign up
-    Please select a role
-    """
+    Then I should see following messages on sign up page:
+      | type  | message                       |
```
тут мався на увазі не тип, а поле, до якого відноситься помилка. Це тільки у випадку інлайн помилок, якщо вони десь всі разом, то перший стобчик зайвий

romikoops commented on features/step_definitions/account_steps.rb:
```diff
@@ -58,4 +48,13 @@
+  res = table.raw.map do |array|
+    array.last if array.first == 'error'
+  end
+  res.compact.each do |str|
```
реалізація не вірна. Потрібно зхватити всі помилки одним махом. Інакше, якщо на формі багато полів і на всіх на них валідації і звалиться перша перевірка, інші не перевіряться

romikoops commented on features/step_definitions/common_steps.rb:
```diff
@@ -52,14 +40,13 @@
 
 Then /^I should be logged in the system$/ do
-  expect(DashboardPage).to be_authenticated
+  DashboardPage.on { has_no_main_menu_section? }
```
це не перевірка, потрібно використовувати rspec predicate matchers. Почитай ще раз їх

```ruby
DashboardPage.on { is_expected.to have_no_main_menu_section }
```

Commit:
PR fixes

Corrections according to https://github.com/mikhaillopandia/howitzer_cucumber_freelancer/pull/1

romikoops commented on features/step_definitions/account_steps.rb:
```diff
@@ -10,45 +10,35 @@
 
-When /^I select (.+) on (.+) page$/ do |what, page|
+When /^I select (.+) on (.+) page$/ do |_what, page|
   page.on { looking_to_hire_radio_element.click }
```
Пагано. зроби метод, який обирає щось, а в середині робить те що ти написав

```ruby
page.on { choose_xxx }
```

```ruby
class XPage
   def choose_xxx
      looking_to_hire_radio_element.click
   end
end
```

romikoops commented on features/step_definitions/account_steps.rb:
```diff
@@ -10,45 +10,35 @@
 end
 
-When /^I select (.+) on (.+) page$/ do |what, page|
+When /^I select (.+) on (.+) page$/ do |_what, page|
```
якщо what ти не використовуєш, тоді напиши конкретику там I select xxx on (.+) ...

Commit:
Fixes fo PR
